core/src/main/java/com/statful/client/core/buffer/StandardBuffer.java

```diff
@@ -48,13 +48,17 @@ public final boolean addToBuffer(final String metric) {
      * @return A {@link String} with all the metrics
      */
     public final String readBuffer() {
-        Collection<String> messages = new ArrayList<String>();
+        Collection<String> messages = new ArrayList<>();
         buffer.drainTo(messages, flushSize);
+        String metricDelimiter = "";
 
         StringBuilder sb = new StringBuilder();
+
         for (String metric : messages) {
-            sb.append(metric).append("\n");
+            sb.append(metricDelimiter).append(metric);
+            metricDelimiter = "\n";
```

Comment:
I'm not the biggest fan of this approach since it makes the code harder to read and has an assignment on every loop operation.

Not great also, but what do you think of something like this:

    for (String metric : messages) {

        sb.append(metric).append("\n");

    }

    sb.setLength(sb.length() - 1);
    return sb.toString()


?